### ISSUE TITLE
[TASK] Add GitHub action to run the unit tests

### ISSUE THREAD

**Issue by jmakhack** [jmakhack/myanimelist-cli#21]
## Task Context

After the work for #13 has been rebased and merged into the master branch, include the running of these unit tests as a part of the GitHub actions pipeline that must pass for every pull request. The tests should be runnable via the `make test` command.

## Additional Context

For any questions or discussions, join the [Discord](https://discord.gg/RTgxfFW9mS) to meet and engage with other contributors!

[![Discord](https://img.shields.io/discord/1023562225790767175?color=%20%237289DA&label=discord&logo=discord&logoColor=%20%237289DA)](https://discord.gg/RTgxfFW9mS)

**Comment by VividhPandey003** [comment]
Hey, I am a beginner to Open Source and really looking forward to contribute to this repo. Could you please tell the tech stack required for this issue and if I'm able to, assign this issue to me

**Comment by jmakhack** [comment]
Hey @VividhPandey003, actually #29 needs to be completed before this one. You can pick that issue up if you're interested or look into a separate one.

All the code in this project is essentially written in C. https://github.com/jmakhack/myanimelist-cli/blob/master/CONTRIBUTION.md has more details on the libraries that is used in this project.
